appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/WalletUtils.java

```diff
@@ -40,63 +54,111 @@ public static boolean hasWalletInstalled() {
+    channelHeadUp.setDescription(description);
+    notificationManager.createNotificationChannel(channelHeadUp);
   }
 
-  public static Activity getActivity() throws Exception {
-    Class activityThreadClass = Class.forName("android.app.ActivityThread");
-    Object activityThread = activityThreadClass.getMethod("currentActivityThread")
-        .invoke(null);
-    Field activitiesField = activityThreadClass.getDeclaredField("mActivities");
-    activitiesField.setAccessible(true);
-    ArrayMap activities = (ArrayMap) activitiesField.get(activityThread);
-    for (Object activityRecord : activities.values()) {
-      Class activityRecordClass = activityRecord.getClass();
-      Field pausedField = activityRecordClass.getDeclaredField("paused");
-      pausedField.setAccessible(true);
-      if (!pausedField.getBoolean(activityRecord)) {
-        Field activityField = activityRecordClass.getDeclaredField("activity");
-        activityField.setAccessible(true);
-        Activity activity = (Activity) activityField.get(activityRecord);
-        return activity;
-      }
+  private static Intent getNotificationIntent() {
+    Intent intent = null;
+    intent = new Intent(Intent.ACTION_VIEW, Uri.parse("market://details?id=" + walletPackageName));
+    intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
+
+    if (hasAptoideInstalled()) {
+      intent.setPackage(aptoidePackageName);
+    }
+
+    PackageManager packageManager = context.getPackageManager();
+    ApplicationInfo applicationInfo = null;
+    Resources resources = null;
+
+    try {
+      applicationInfo =
+          packageManager.getApplicationInfo(context.getPackageName(), PackageManager.GET_META_DATA);
+      resources = packageManager.getResourcesForApplication(applicationInfo);
+    } catch (PackageManager.NameNotFoundException e) {
+      e.printStackTrace();
+      return null;
+    }
+
+    int applicationIconResID = applicationInfo.icon;
+
+    String iconName = resources.getResourceName(applicationIconResID);
+    String typeName = resources.getResourceTypeName(applicationIconResID);
+    String packageName = resources.getResourcePackageName(applicationIconResID);
+    int identifier = resources.getIdentifier(iconName, typeName, packageName);
+
+    intent.putExtra("identifier", identifier);
+
+    return intent;
+  }
+
+  private static Notification buildNotification(String channelId, Intent intent,
+      boolean useTimeOut) {
+
+    Notification.Builder notBuilder = null;
+
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+      notBuilder = new Notification.Builder(context, channelId);
+    } else {
+      notBuilder = new Notification.Builder(context);
     }
-    return null;
+
+    pendingIntent = PendingIntent.getActivity(context, 0, intent, 0);
+
+    notBuilder.setSmallIcon(intent.getExtras()
+        .getInt("identifier"))
+        .setContentTitle("You need the AppCoins Wallet!")
```

Comment:
I think you forgot to use the texts you created in the resources strings.xml
here